RadixWallet/Core/FeaturePrelude/AddressView/AddressDetails+View.swift

```diff
@@ -0,0 +1,266 @@
+			VStack(spacing: .medium2) {
+				VStack(spacing: .small2) {
+					Text(L10n.AddressDetails.fullAddress)
+						.foregroundColor(.app.gray1)
+					Text(address)
+				}
+				actions
+			}
+			.textStyle(.body1Header)
+			.padding(.vertical, .medium1)
+			.padding(.horizontal, .medium3)
+			.background(Color.app.gray5)
+			.cornerRadius(.small1)
+		}
+
+		private var actions: some SwiftUI.View {
+			HStack(spacing: .large3) {
+				Button(L10n.AddressDetails.copy, image: .copy) {
+					store.send(.view(.copyButtonTapped))
+				}
+				Button(L10n.AddressDetails.enlarge, image: .fullScreen) {
+					store.send(.view(.enlargeButtonTapped))
+				}
+				Button(L10n.AddressDetails.share, systemImage: "square.and.arrow.up") {
+					store.send(.view(.shareButtonTapped))
+				}
+			}
+			.padding(.horizontal, .medium2)
+			.foregroundColor(.app.gray1)
+		}
+
+		private func bottom(showVerifyOnLedger: Bool) -> some SwiftUI.View {
+			VStack(spacing: .medium3) {
+				Button(L10n.AddressDetails.viewOnDashboard) {
+					store.send(.view(.viewOnDashboardButtonTapped))
+				}
+				.buttonStyle(
+					.secondaryRectangular(
+						shouldExpand: true,
+						trailingImage: .init(.iconLinkOut)
+					)
+				)
+				if showVerifyOnLedger {
+					Button(L10n.AddressDetails.verifyOnLedger) {
+						store.send(.view(.verifyOnLedgerButtonTapped))
+					}
+					.buttonStyle(.secondaryRectangular(shouldExpand: true))
+				}
+			}
+		}
+
+		private func qrCode(qrImage: Loadable<CGImage>) -> some SwiftUI.View {
+			ZStack {
+				switch qrImage {
+				case let .success(value):
+					Image(decorative: value, scale: 1)
+						.resizable()
+						.aspectRatio(1, contentMode: .fit)
+						.transition(.scale(scale: 0.95).combined(with: .opacity))
+				case .failure:
+					Text(L10n.AddressDetails.qrCodeFailure)
+						.textStyle(.body1HighImportance)
+						.foregroundColor(.app.alert)
+				case .loading:
+					ProgressView()
+				case .idle:
+					EmptyView()
+				}
+			}
+			.animation(.easeInOut, value: qrImage.isSuccess)
+		}
+	}
+}
+
+// MARK: - Enlarged view
+private extension AddressDetails.View {
+	func enlargedView(text: AttributedString) -> some SwiftUI.View {
+		Group {
+			Text(text)
+				.textStyle(.enlarged)
+				.foregroundColor(.app.white)
+		}
+		.multilineTextAlignment(.center)
+		.padding(.small1)
+		.background(.app.gray1.opacity(0.8))
+		.cornerRadius(.small1)
+		.padding(.horizontal, .large2)
+		.onTapGesture {
+			store.send(.view(.hideEnlargedView))
+		}
+		.background(enlargedViewBackrgound)
+	}
+
+	private var enlargedViewBackrgound: some View {
+		Color.black.opacity(0.2)
+			.frame(width: UIScreen.main.bounds.width * 2, height: UIScreen.main.bounds.height * 2)
+			.contentShape(Rectangle())
+			.onTapGesture {
+				store.send(.view(.hideEnlargedView))
+			}
+	}
+}
+
+private extension AddressDetails.ViewState {
+	static func colorised(address: LedgerIdentifiable.Address) -> AttributedString {
```

Comment:
I think this should be either a local private function, or, if you might want to use it somewhere else, a `var` on `LedgerIdentifiable.Address` or something.

Reply:
The idea was to computate it only once

Reply:
The ViewState is actually re-computed every time anything changes, as is the entire SwiftUI View, so I don't think it makes any difference. 

But either way, my idea was just to change the ergonomics to something like this:

`let addressString = address.colorizedString`

It might be that Observable is smart enough to only recompute it when the underlying address changes if you do it like that. When it's in the ViewState, any change to any property on the ViewState would recompute everything.

Seems like it could be useful more generally, btw.